pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -49,44 +50,113 @@ const (
 var _ = Framework(&framework{})
 
 // NewFramework initializes plugins given the configuration and the registry.
-func NewFramework(r Registry, _ *runtime.Unknown) (Framework, error) {
+func NewFramework(r Registry, plugins *config.Plugins, args []config.PluginConfig) (Framework, error) {
 	f := &framework{
 		registry:         r,
 		nodeInfoSnapshot: cache.NewNodeInfoSnapshot(),
 		plugins:          make(map[string]Plugin),
 		waitingPods:      newWaitingPodsMap(),
 	}
+	if plugins == nil {
+		return f, nil
+	}
+
+	// get needed plugins from config
+	pg := pluginsNeeded(plugins)
+	if len(pg) == 0 {
+		return f, nil
+	}
 
-	// TODO: The framework needs to read the scheduler config and initialize only
-	// needed plugins. In this initial version of the code, we initialize all.
+	pluginConfig := pluginNameToConfig(args)
 	for name, factory := range r {
-		// TODO: 'nil' should be replaced by plugin config.
-		p, err := factory(nil, f)
+		// initialize only needed plugins
+		if _, ok := pg[name]; !ok {
+			continue
+		}
+
+		// find the config args of a plugin
+		pc := pluginConfig[name]
```

Comment:
Should we check that the plugin has corresponding config ?

Reply:
For now, I think it's okay to pass nil config into plugin factories. I'm not sure it would help to pass in an "empty" `runtime.Unknown` object.